agents/executor/executor.go

```diff
@@ -110,3 +106,12 @@ func inputsToString(inputValues map[string]any) (map[string]string, error) {
 
 	return inputs, nil
 }
+
+func getNameToTool(t []tools.Tool) map[string]tools.Tool {
+	nameToTool := make(map[string]tools.Tool, len(t))
```

Comment:
nit:
```suggestion
	if len(t) == 0 {
		return nil
	}

	nameToTool := make(map[string]tools.Tool, len(t))
```

Reply:
You're correct, it doesn't really make a difference, it's just a [style choice](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices). I marked it as nit, feel free to ignore it 😊